Templates/colors.json

```diff
@@ -0,0 +1,4687 @@
+      ]
+    },
+    "ObjShadowSpawner": {
+      "fields": [
+        { "name": "Width", "type": "float32" },
+        { "name": "Height", "type": "float32" },
+        { "name": "Depth", "type": "float32" }
+      ]
+    },
+    "TestObjSpawner": {
+      "fields": [
+      ]
+    },
+    "CCTTestObjSpawner": {
+      "fields": [
+      ]
+    },
+    "ObjAppearThornBallSpawner": {
+      "fields": [
+        { "name": "Time", "type": "float32" },
+        { "name": "Distance", "type": "float32" }
+      ]
+    },
+    "ObjChainSpikeBallSpawner": {
+      "fields": [
+        { "name": "Radius", "type": "float32" },
+        { "name": "AngularSpeed", "type": "float32" },
+        { "name": "Phase", "type": "float32" },
+        { "name": "RotateType", "type": "ObjChainSpikeBallSpawner::RotateType" }
+      ]
+    },
+    "ObjChaserBombSpawner": {
+      "fields": [
+        { "name": "ThresholdStubmle", "type": "float32" },
+        { "name": "IsBigChaserMode", "type": "bool" }
+      ]
+    },
+    "ObjDirectionalThornSpawner": {
+      "fields": [
+        { "name": "OffTime", "type": "float32" },
+        { "name": "OnTime", "type": "float32" },
+        { "name": "MoveTime", "type": "float32" },
+        { "name": "ForetasteTime", "type": "float32" },
+        { "name": "Phase", "type": "float32" },
+        { "name": "TargetID", "type": "object_reference" },
+        { "name": "UsePanel", "type": "bool" },
+        { "name": "Type", "type": "ObjDirectionalThornSpawner::Type" }
+      ]
+    },
+    "ObjFlameSpawner": {
+      "fields": [
+        { "name": "OffTime", "type": "float32" },
+        { "name": "OnTime", "type": "float32" },
+        { "name": "LengthType", "type": "ObjFlameSpawner::LengthType" },
+        { "name": "Phase", "type": "float32" },
+        { "name": "Type", "type": "ObjFlameSpawner::Type" }
+      ]
+    },
+    "ObjGrindThornSpawner": {
+      "fields": [
+      ]
+    },
+    "ObjMovingThornBallSpawner": {
+      "fields": [
+        { "name": "Time", "type": "float32" },
+        { "name": "Distance", "type": "float32" },
+        { "name": "Phase", "type": "float32" },
+        { "name": "Direction", "type": "ObjMovingThornBallSpawner::Direction" }
+      ]
+    },
+    "ObjRollingBallSpawner": {
+      "fields": [
+        { "name": "Radius", "type": "float32" },
+        { "name": "Cycle", "type": "float32" },
+        { "name": "Phase", "type": "float32" },
+        { "name": "BallCount", "type": "uint8" },
+        { "name": "Option", "type": "ObjRollingBallSpawner::Option" }
+      ]
+    },
+    "ObjThornBallSpawner": {
+      "fields": [
+      ]
+    },
+    "ObjThornCylinderSpawner": {
+      "fields": [
+        { "name": "IsLongModel", "type": "bool" }
+      ]
+    },
+    "ObjEnvSoundSpawner": {
+      "fields": [
+        { "name": "CueID", "type": "uint32" },
+        { "name": "Volume", "type": "float32" },
+        { "name": "IsPlay3D", "type": "bool" }
+      ]
+    },
```

Comment:
Parameters named based on in-game functionality and parameters from other sound related objects. Set data parser function (0x8021F84C, NTSC-J Wii) and `AddCallback` function (0x8021F7E4, NTSC-J Wii) used for additional help.